stdlib/filename.mli

```diff
@@ -46,8 +46,10 @@ val check_suffix : string -> string -> bool
 
 val chop_suffix : string -> string -> string
 (** [chop_suffix name suff] removes the suffix [suff] from
-   the filename [name]. The behavior is undefined if [name] does not
-   end with the suffix [suff]. *)
+   the filename [name].
+
+   @since 4.08 If [name] does not end with the suffix [suff],
```

Comment:
If one wants to fully specify the behavior, one should explain what "ending with the suffix" means (also for other existing functions).  This is not straightforward: for Windows ports, the comparison is case-insensitive on ASCII characters; but not, for instance on macOS (whose file-system is also usually case-insensitive, I believe).

Reply:
I've specified `chop_suffix` in terms of `check_suffix`. Your suggestion could be done as a separate patch if there is desire to do so.